File: frontend/types/document.ts
export type DocumentType =
  | "GST_CERTIFICATE"
  | "UDYAM_CERTIFICATE"
  | "PAN_CARD"
  | "BANK_STATEMENT"
  | "ITR_ACKNOWLEDGEMENT"
  | "OTHER";

export type DocumentStatus = "PENDING" | "PROCESSING" | "COMPLETED" | "FAILED";

export type DocumentRecord = {
  id: number;
  business_id: number;
  document_type: DocumentType | string;
  status: DocumentStatus | string;
  original_filename: string;
  stored_filename: string;
  file_path: string;
  mime_type: string | null;
  file_size_bytes: number;
  uploaded_at: string;
  processed_at: string | null;
};

export type DocumentExtraction = {
  document_id: number;
  business_id: number;
  document_type: string;
  status: string;
  extraction_status: string;
  confidence_score: number | null;
  extracted_fields: Record<string, unknown>;
  processed_at: string | null;
};

export type DocumentExtractionList = {
  extractions: DocumentExtraction[];
};

File: backend/app/schemas/document.py
from datetime import datetime
from typing import List, Literal, Optional

from pydantic import BaseModel, ConfigDict

DocumentType = Literal[
    "GST_CERTIFICATE",
    "UDYAM_CERTIFICATE",
    "PAN_CARD",
    "BANK_STATEMENT",
    "ITR_ACKNOWLEDGEMENT",
    "OTHER",
]

DocumentStatus = Literal["PENDING", "PROCESSING", "COMPLETED", "FAILED"]


class DocumentCreate(BaseModel):
    business_id: int
    document_type: DocumentType


class DocumentRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    business_id: int
    document_type: str
    status: str
    original_filename: str
    stored_filename: str
    file_path: str
    mime_type: Optional[str]
    file_size_bytes: int
    uploaded_at: datetime
    processed_at: Optional[datetime]


class DocumentList(BaseModel):
    documents: List[DocumentRead]
